### ISSUE TITLE
Prommise is rejected every time I login in android

### ISSUE THREAD

**Issue by Adarsh077** [CodetrixStudio/CapacitorGoogleAuth#70]
E/Capacitor/Plugin: Something went wrong
    com.google.android.gms.common.api.ApiException: 10: 
        at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus(com.google.android.gms:play-services-base@@17.1.0:4)
        at com.google.android.gms.auth.api.signin.GoogleSignIn.getSignedInAccountFromIntent(com.google.android.gms:play-services-auth@@18.1.0:9)
        at com.codetrixstudio.capacitor.GoogleAuth.GoogleAuth.handleOnActivityResult(GoogleAuth.java:70)
        at com.getcapacitor.Bridge.onActivityResult(Bridge.java:776)
        at com.getcapacitor.BridgeActivity.onActivityResult(BridgeActivity.java:214)
        at android.app.Activity.dispatchActivityResult(Activity.java:8320)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5186)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5234)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:140)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:100)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2230)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:7810)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1027)

This is the error shown in android studio console

**Comment by ArturV93** [comment]
This comment should help you to solve issue
https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-711376468

**Comment by aren1989** [comment]
> This comment should help you to solve issue
> [#67 (comment)](https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-711376468)

I have same issue, I try your solution but it doesn't work

**Comment by ArturV93** [comment]
> > This comment should help you to solve issue
> > [#67 (comment)](https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-711376468)
> 
> I have same issue, I try your solution but it doesn't work
 
Check if your SHA-1 is right? also try enable the Google Signin in Firabase project
![image](https://user-images.githubusercontent.com/10709971/99531542-cd3d9680-299a-11eb-92cd-82c476a23251.png)

**Comment by Adarsh077** [comment]
> This comment should help you to solve issue
> [#67 (comment)](https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-711376468)



> > > This comment should help you to solve issue
> > > [#67 (comment)](https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-711376468)
> > 
> > 
> > I have same issue, I try your solution but it doesn't work
> 
> Check if your SHA-1 is right? also try enable the Google Signin in Firabase project
> ![image](https://user-images.githubusercontent.com/10709971/99531542-cd3d9680-299a-11eb-92cd-82c476a23251.png)

I check all the above parameters but the error is still coming.
I have also referred to this tutorial https://www.youtube.com/watch?v=Rs1imvTbeN0

**Comment by Adarsh077** [comment]
@ArturV93 @aren1989 

https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/67#issuecomment-712786126

I build a signed apk as mentioned in this comment, and it worked.

**Comment by Adarsh077** [comment]
I am closing this issue. I think this should be mentioned in the documentation also.

**Comment by Adarsh077** [comment]
I would also like to mention that instead of web client ID I used Android Client Id in capacitor.confiig.json

**Comment by tylerku** [comment]
@Adarsh077 I have an Android Client ID for both my debug version and production version of the app. If I use the Android ID won't it then be the wrong ID when I put it on the Google Play Store as a signed apk using my production keystore?

**Comment by Adarsh077** [comment]
@tylerku I used web client ID which is provided in Google play console

**Comment by tylerku** [comment]
@Adarsh077 Oh I totally missed your edit. Thanks! 

So I'm getting my promise rejected as well with and `com.google.android.gms.common.api.ApiException: 10:` in the console when I call `GoogleAuth.signIn()`. I've gone through all the issues I can find on the subject and still haven't had any luck. Here is what I've done. 

1) Added my debug and production SHA-1 fingerprints to my firebase project (which auto generates them in the corresponding Google Console project). 
2) Added the `google-services.json` file to the root level of my project.
3) Added all the dependencies to my build.gradle that the firebase docs mentions.. as well as the ones that the google sign in docs mention
`  
    dependencies {
        ....
        implementation platform('com.google.firebase:firebase-bom:26.5.0')
        implementation 'com.google.firebase:firebase-analytics'
        implementation 'com.google.firebase:firebase-auth'
        implementation 'com.google.android.gms:play-services-auth:19.0.0'
    }
`
4) Added the other code snippets as explained by Firebase console when adding an android app.
5) Followed all the steps in the README (always using the web client id from google console as the client id referred to in the README steps)
6) Added a support email address in both Google console and Firebase console.
7) Turned on Google Sign In in the Firebase project settings 

I'm really at a loss as to what else I can try to get rid of this `ApiException: 10`. I've read that it usually means that my project is not configured correctly, but I've looked over the configuration multiple times and it seems right to me so I thought I'd ask for help. Do you have any idea what I may be missing?

**Comment by Adarsh077** [comment]
@tylerku Try building a signed apk and then try to login.

**Comment by tylerku** [comment]
@Adarsh077 I tried that and no luck. Building both generated signed apk and app bundle both don't work.. as well as the debug version signed with the debug key

**Comment by ArturV93** [comment]
1) Are you generated Web Client token?
2) Add Web Client id everywhere following the README document 
3) is your google service in the folder? (Project name)\android\app\google-services.json
4) Check the google-service.json if All keys there Andoird, Ios and web 
5) I assume if you followed the document you added web client id in( android\app\src\main\res\xml, android\app\src\main\res\values\strings.xml)

6) https://medium.com/fabcoding/how-to-obtain-sha1-keys-for-debug-and-release-android-studio-mac-34ff3703470b 
Follow the debug key, then go to https://console.developers.google.com/apis/  find your project and compare the key 

Can you build in ios? 
Is your code has public access?

**Comment by tylerku** [comment]
I went over all those things and discovered my issue. I had only uploaded the `Upload Key` SHA-1 certificate fingerprint to my firebase project. I also had to enter the `App Signing Key` SHA-1 certificate fingerprint to my firebase project (found in the Google Play console under `Settings` > `App Integrity`). Since Google signs each of the app releases with that `App Signing Key` when it is released on the store, if a developer doesn't upload the `App Signing Key` SHA-1 fingerprint, then the released version of the app won't work when users try to login. 

@ArturV93 thanks for asking me all those questions because they helped me discover the root of my problem. 
Hopefully this can help someone in the future that runs into a similar issue :)
